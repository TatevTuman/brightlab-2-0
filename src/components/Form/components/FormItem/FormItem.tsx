import React, { memo } from 'react'
import { Children } from '@types'
import styles from '../../Form.module.scss'

export interface FormItemProps {
  children: Children
  justify?: string
}

const FormItem: React.FC<FormItemProps> = ({ children, justify }) => {
  const justifyContent = justify

  return (
    <div className={styles.formItem} style={{ justifyContent }}>
      {children}
    </div>
  )
}

export default memo(FormItem)
