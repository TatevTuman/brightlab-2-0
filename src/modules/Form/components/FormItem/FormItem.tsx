import React, { memo } from 'react'
import { Children } from '@types'
import styles from '../../Form.module.scss'

export interface FormItemProps {
  children: Children
  justify?: string
  margin?: string
}

const FormItem: React.FC<FormItemProps> = props => {
  const { children, justify, margin } = props
  const justifyContent = justify

  return (
    <div className={styles.formItem} style={{ justifyContent, margin }}>
      {children}
    </div>
  )
}

export default memo(FormItem)
