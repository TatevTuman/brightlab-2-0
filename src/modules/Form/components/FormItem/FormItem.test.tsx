import React from 'react'
import { cleanup, waitFor, render } from '@testing-library/react'
import FormItem from './FormItem'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('FormItem', () => {
  it('renders correctly', async () => {
    const { container, getByText } = render(<FormItem>children</FormItem>)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
    expect(getByText('children')).toBeInTheDocument()
  })
})
