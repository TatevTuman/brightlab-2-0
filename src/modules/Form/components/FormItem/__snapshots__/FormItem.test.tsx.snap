// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItem renders correctly 1`] = `
<div>
  <div
    class="form-item"
  >
    children
  </div>
</div>
`;
